### DISCUSSION TITLE
Turborepo / PNPM setup for libraries with local packages

### DISCUSSION THREAD

**Discussion by NicHaley** [vercel/turborepo#1510] [Help] [answered]
I'm using Turborepo with PNPM and having a difficult time configuring the following scenario:

- Monorepo contains two packages: `foo@1.0.0` and `bar@1.0.0`
- The `foo` package is a library which is published to NPM
- The `bar` package is a private library which is not published to NPM. (I don't want this package to be published)
- `foo` depends on `bar`
- `foo` uses the PNPM Workspace protocol to link to bar, ie. `"bar": "workspace: *"`
- Locally, everything runs great. But when I publish `foo`, `bar` will try to point to an external package (if one exists with the same name) on the NPM registry. What I want to happen is for `bar` to be bundled with `foo`. I have tried using [bundledDependencies](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#bundleddependencies), but without any luck.

**Comment by cdaringe** [comment, marked as answer]
hey @NicHaley, i'm pretty sure you'll need to bring-your-own build step.

> What I want to happen is for bar to be bundled with foo

AFAIK, turbo nor (p)npm provide solutions for this. solutions like rollup/webpack/esbuild/tsup etc help precisely with this type of use case, but I cannot offer you an off-the-shelf recipe. surely a variety of solutions exist out there, but i wouldn't look to the package manager or the task runner _first_.

good luck out there, please share what you end up doing!
